### DISCUSSION TITLE
Multiple split & run commands on keypress

### DISCUSSION THREAD

**Discussion by elazar125** [wezterm/wezterm#2728] [Q&A] [answered]
Hello,

I've been fighting with this for a while now and I'm hoping someone can help. I want to run four different commands in specific directories all at once on a keypress, in order to spin up a complex application.

I've found similar discussions (#2388, #2520, #2397), but they haven't helped me figure it out.

The current behaviour is that it creates a new tab, opens the first 2/4 panes, starts running the commands, then they crash and the panes close. The last two panes never appear. The commands themselves don't crash when run manually. 

Code is as follows:
```lua
 {key="p", mods="ALT|SHIFT", action=wezterm.action_callback(function(win, pane)
     local new_tab, first_pane, new_win = win:mux_window():spawn_tab {
         cwd="C:\\Path\\To\\Project1,
         args={"dotnet", "watch", "run"},
     }
     first_pane:split {
         direction="Right",
         size=0.25,
         cwd="C:\\Path\\To\\Project2",
         args={"dotnet", "watch", "run"},
     }
     first_pane:split {
         direction="Right",
         size=0.33,
         cwd="C:\\Path\\To\\Project1",
         args={"grunt", "dev"},
     }
     first_pane:split {
         direction="Right",
         size=0.5,
         cwd="C:\\Path\\To\\Project2",
         args={"grunt", "dev"},
     }
end)},
```

I've also previously tried using `wezterm.action.Multiple` to wrap callls to `wezterm.action.SpawnCommandInNewTab` and `wezterm.action.SplitPane`. Doing it that way, the first two panes opened and ran just fine (no crash or auto-close), the last two never showed up. 

In case it's relevant, I'm running wezterm version 20220905-102802-7d4b8249 on Windows 10.

Thanks in advance for your help!

**Comment by wez** [comment, marked as answer]
Please update to a nightly build and see if that resolves the issue.
You can obtain a nightly build using the "Nightly" buttons on https://wezfurlong.org/wezterm/install/windows.html
If the issue persists, please open an issue: the form for that includes instructions on gathering logs that will help to understand what is happening.

**Reply by elazar125** [reply]
Thanks! The location of the logs helped a bunch, I've opened Issue #2744 for this bug.

I've found a workaround in the meantime that I'll post here in case anyone else comes across this, it's to use pane:send_text() to run the command.

Example code as follows:
```lua
{key="p", mods="ALT|SHIFT", action=wezterm.action_callback(function(win, pane)
    local new_tab, first_pane, new_win = win:mux_window():spawn_tab {
        cwd="C:\\Path\\To\\Project1",
    }
    local second_pane = first_pane:split {
        direction="Right",
        size=0.25,
        cwd="C:\\Path\\To\\Project2",
    }
    local third_pane = first_pane:split {
        direction="Right",
        size=0.33,
        cwd="C:\\Path\\To\\Project1",
    }
    local fourth_pane = first_pane:split {
        direction="Right",
        size=0.5,
        cwd="C:\\Path\\To\\Project2",
    }
    first_pane:send_text("dotnet watch run\r")
    second_pane:send_text("dotnet watch run\r")
    third_pane:send_text("grunt dev\r")
    fourth_pane:send_text("grunt dev\r")
end)},
```
